Beam.Server.Application/Services/UserService.cs

```diff
@@ -0,0 +1,70 @@
+﻿using Beam.Application.Services.Interfaces;
+using Beam.Infrastructure;
+using Beam.Infrastructure.Entities;
+using Microsoft.EntityFrameworkCore;
+
+namespace Beam.Application.Services;
+
+public class UserService : IUserService
+{
+    private readonly BeamDbContext _dbContext;
+
+    public UserService(BeamDbContext dbContext)
+    {
+        _dbContext = dbContext;
+    }
+
+    public async Task<List<User>> GetAllAsync()
+    {
+        var users = await _dbContext.Users.ToListAsync();
+        return users;
+    }
+
+    public async Task<User>GetByIdAsync(Guid id)
+    {
+        var user = await _dbContext.Users.FirstOrDefaultAsync(u => u.Id == id);
+        if (user == null)
+        {
+            throw new InvalidOperationException("User not found");
```

Comment:
You have `NotFoundException` in `Beam.Server.Core` project, that you can use here;
```
throw new NotFoundException($"User with Id {id} not found");
```